ironfish/src/network/peerNetwork.ts

```diff
@@ -518,21 +527,26 @@ export class PeerNetwork {
     peer: Peer,
     gossipMessage: GossipNetworkMessage,
   ): Promise<void> {
-    if (!this.seenGossipFilter.added(gossipMessage.nonce)) {
-      return
-    }
-
     const peerIdentity = peer.getIdentityOrThrow()
 
     if (gossipMessage instanceof NewBlockMessage) {
+      if (!this.seenGossipFilter.added(gossipMessage.nonce)) {
```

Comment:
This is just continuing to use the `seenGossipFilter` for blocks. I didn't want to also refactor block logic for this proposal. In this new logic `seenGossipFilter` will only contain block nonces